packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2119,6 +2134,243 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     );
   }
 
+  async indexEntitiesWithDependencyIntoGraph<T extends V1_PackageableElement>(
+    graph: PureModel,
```

Comment:
I think we should pre-filter the elements to improve performance here.

As discussed, We could even tinker with the entities (JSON objects) to lessen the load for deserialization, maybe could you add a small comment about that? It's clearly premature optimization, but worth noting.